src/pages/workspaces/workspace/DeleteWorkspaceModal.ts

```diff
@@ -0,0 +1,167 @@
+import _ from 'lodash/fp';
+import pluralize from 'pluralize';
+import { useState } from 'react';
+import { b, div, h, label, p, span } from 'react-hyperscript-helpers';
+import { ButtonPrimary, Link, spinnerOverlay } from 'src/components/common';
+import { icon } from 'src/components/icons';
+import { TextInput } from 'src/components/input';
+import Modal from 'src/components/Modal';
+import { bucketBrowserUrl } from 'src/libs/auth';
+import colors from 'src/libs/colors';
+import { warningBoxStyle } from 'src/libs/style';
+import * as Utils from 'src/libs/utils';
+import { isGoogleWorkspace, WorkspaceWrapper as Workspace } from 'src/libs/workspace-utils';
+import { useDeleteWorkspaceState } from 'src/pages/workspaces/workspace/useDeleteWorkspaceState';
+
+interface DeleteWorkspaceModalProps {
+  workspace: Workspace;
+  onDismiss: () => void;
+  onSuccess: () => void;
+}
+
+const DeleteWorkspaceModal = (props: DeleteWorkspaceModalProps) => {
+  const [deleteConfirmation, setDeleteConfirmation] = useState('');
+  const { workspace, onDismiss, onSuccess } = props;
+  const {
+    workspaceResources,
+    loading,
+    deleting,
+    isDeleteDisabledFromResources,
+    workspaceBucketUsageInBytes,
+    collaboratorEmails,
+    hasApps,
+    deleteWorkspace,
+  } = useDeleteWorkspaceState({ workspace, onDismiss, onSuccess });
+
+  const getStorageDeletionMessage = () => {
+    return div({ style: { marginTop: '1rem' } }, [
+      'Deleting it will delete the associated ',
+      isGoogleWorkspace(props.workspace)
+        ? h(
+            Link,
+            {
+              ...Utils.newTabLinkProps,
+              href: bucketBrowserUrl(props.workspace.workspace.bucketName),
+            },
+            ['Google Cloud Bucket']
+          )
+        : 'Azure Storage Container',
+      ' and all its data',
+      workspaceBucketUsageInBytes !== undefined &&
+        span({ style: { fontWeight: 600 } }, [` (${Utils.formatBytes(workspaceBucketUsageInBytes)})`]),
+      '.',
+    ]);
+  };
+
+  const getResourceDeletionMessage = () => {
+    // On Azure, JupyterLab is returned as a runtime instead of an app (on GCP, Jupyter seems to be neither).
+    // WDS and Cromwell are apps.
+    const nNonDeletableResources =
+      (workspaceResources?.nonDeleteableApps.length ?? 0) + (workspaceResources?.nonDeleteableRuntimes.length ?? 0);
+    const nDeletableResources =
+      (workspaceResources?.deleteableApps.length ?? 0) + (workspaceResources?.deleteableRuntimes.length ?? 0);
+    const nonDeletableResourcesVerb = nNonDeletableResources === 1 ? 'is' : 'are';
+    const nonDeletableResourceMessage = `You cannot delete this workspace because there ${nonDeletableResourcesVerb} ${pluralize(
+      'application',
+      nNonDeletableResources,
+      true
+    )} that ${nonDeletableResourcesVerb} currently not deletable. Only applications in ('ERROR', 'RUNNING') status can be automatically deleted.`;
```

Comment:
Some screenshots showing the above message:

1) WDS running (deletable) and JupyterLab starting up:

![image](https://github.com/DataBiosphere/terra-ui/assets/484484/87a921eb-f64f-4d2d-88b4-d568eb0b862c)

2) Cromwell and WDS both starting up:

![image](https://github.com/DataBiosphere/terra-ui/assets/484484/d8b73c0e-a33a-4bcb-ba4c-342fa2a6ac32)